www/js/coolhole-util.js

```diff
@@ -301,3 +320,166 @@ function playSound( sfxLibItem ) {
+            let prevOverlapEmoteTitle = "";                                 // used to detect repeating overlapping emotes
+            let emoteCombo = 0;                                             // used to offset repeating overlapping emotes a bit each time
+
+            for(var i = 0; i < emotesFound.length; i++) {
+
+                // If its an overlapping emote, proceed to reposition it.
+                if(emotesFound[i].classList.contains("emote-overlap")) {
+                    let emoteTitle = emotesFound[i].getAttribute("title");
+                    let overlapEmoteRect = emotesFound[i].getBoundingClientRect();
+                    let anchor = OVERLAP_EMOTES.find(x => x.name === emoteTitle)?.anchor ?? "center";
+
+                    // Make the overlapping emote visible again
+                    emotesFound[i].style.visibility = "";
+
+                    // If the overlapping emote repeats, increase the combo
+                    if(prevOverlapEmoteTitle === emoteTitle)
+                        emoteCombo+=2;
+                    else
+                        emoteCombo = 0;
+
+                    // Just a sanity check
+                    if(prevNormalRect !== null) {
+                        // Adjust the overlap depending on the anchor
+                        switch(anchor) {
+                            case "left": {
+                                let leftPrevEmote = prevNormalRect.left - chatTextRect.left;
+                                emotesFound[i].style.left = leftPrevEmote - Math.round(overlapEmoteRect.width/4) - emoteCombo + "px";
+                                emotesFound[i].style.bottom = -Math.round(overlapEmoteRect.height/2) - emoteCombo + "px";
+                                break;
+                            }
+                            case "right": {
+                                let rightPrevEmote = prevNormalRect.right - chatTextRect.left;
+                                emotesFound[i].style.left = rightPrevEmote - Math.round(overlapEmoteRect.width*3/4) + emoteCombo + "px";
+                                emotesFound[i].style.bottom = -Math.round(overlapEmoteRect.height/2) - emoteCombo + "px";
+                                break;
+                            }
+                            case "center":
+                            default: {
+                                let leftPrevEmote = prevNormalRect.left - chatTextRect.left;
+                                let topPrevEmote = prevNormalRect.top - chatTextRect.top;
+                                emotesFound[i].style.left = leftPrevEmote - Math.round((overlapEmoteRect.width - prevNormalRect.width) / 2) + emoteCombo + "px";
+                                emotesFound[i].style.top = topPrevEmote - Math.round((overlapEmoteRect.height - prevNormalRect.height) / 2) + emoteCombo + "px";
+                                break;
+                            }
```

Comment:
praise: :shipit: I am not going to pretend I even tried to understood the math here and how this shifts the images properly. If it works, no sense in touching it. If there's a better way of doing this, I'm certainly not aware of it lol

Reply:
I don't want to redo the math either haha.
It just satisfies whatever placement I had in my head 3 years ago (or whenever I made the returnFire a feature)